aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConflictHandler.java

```diff
@@ -16,23 +16,288 @@
 package com.amplifyframework.datastore;
 
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.util.ObjectsCompat;
 
 import com.amplifyframework.core.Consumer;
 import com.amplifyframework.core.model.Model;
 
+import java.util.Objects;
+
 /**
- * Handles conflicts between models observed during model synchronization in the DataStore.
- * Such conflicts may arise between version of models kept locally, and competing versions
- * found in the remote AppSync system.
+ * A hook to handle a conflict between local and remote copies of a model.
+ * The DataStore customer can implement their own version of this handler,
+ * and provide that handler to the {@link DataStoreConfiguration} while constructing
+ * the DataStore plugin using
+ * {@link AWSDataStorePlugin#AWSDataStorePlugin(DataStoreConfiguration)}.
  */
 public interface DataStoreConflictHandler {
     /**
-     * Resolves a DataStore conflict.
-     * @param conflictData Data about the conflict.
-     * @param onResult A callback that should be invoked when the conflict handling is complete.
-     * @param <T> The type of model for which a conflict was observed
+     * Factory to obtain a handler that always applies the remote copy
+     * of the data.
+     * @return A DataStore conflict handler that always applies the remote
+     *         copy of the data.
+     */
+    static DataStoreConflictHandler alwaysApplyRemote() {
```

Comment:
Would it make sense to have the inverse of this, `alwaysApplyLocal()`?

Reply:
There was a lot of deliberation on the naming back in April/May when iOS implemented this functionality. At this point, I'm just aligning to the other platforms.

The key difference is that "apply remote" just patches the local data, but doesn't talk to AppSync again. On the other hand, "retry local" will attempt another mutation, using the server's `_version`, but with the local data. If _that_ mutation fails, the overall transaction fails.

^ Maybe we can get this explanation into the docs site, somewhere.